### ISSUE TITLE
ColumnUnknownException thrown when referencing known object keys of objects from system tables

### ISSUE THREAD

**Issue by amotl** [crate/crate#17102]
### CrateDB version

CrateDB Nightly, build dffe52d.

### CrateDB setup information

Standard Docker, with a custom node attribute `-Cnode.attr.storage=hot`.

```
docker run --rm -it --publish=4200:4200 --publish=5432:5432 --env=CRATE_HEAP_SIZE=4g \
  crate/crate:nightly -Cdiscovery.type=single-node -Cnode.attr.storage=hot
```

```
version[5.10.0-SNAPSHOT], pid[1], build[dffe52d/NA], OS[Linux/6.10.14-linuxkit/amd64], JVM[Eclipse Adoptium/OpenJDK 64-Bit Server VM/23.0.1+11]
```

### Problem description

`test_run_reallocate` in [CrateDB Toolkit's data retention subsystem](https://cratedb-toolkit.readthedocs.io/retention.html) fails with current CrateDB Nightly.

-- https://github.com/crate/cratedb-toolkit/actions/runs/12152377093/job/33888670544#step:6:147

### Steps to Reproduce

```sql
create table t01 (id INT PRIMARY KEY ) partitioned by ("id") WITH ("routing.allocation.require.storage" = 'hot');
```

```sql
insert into t01 (id) values (42);
```

```sql
WITH partitions AS (

            SELECT
                *,
                TRY_CAST(p.values['ts_day'] AS BIGINT) AS partition_value
            FROM
                "information_schema"."table_partitions" AS p
            WHERE 1
                AND p.table_schema = 'doc'
                AND p.table_name = 't01'
                AND p.values['id'] < 50
            ORDER BY
                partition_value ASC

    ),
    partition_allocations AS (
      SELECT DISTINCT s.schema_name AS table_schema,
                      s.table_name,
                      s.partition_ident,
                      n.attributes
      FROM sys.shards s
      JOIN sys.nodes n ON s.node['id'] = n.id
    )
    SELECT *
    FROM partitions p
    JOIN partition_allocations a ON a.table_schema = p.table_schema
      AND a.table_name = p.table_name
      AND p.partition_ident = a.partition_ident
      AND attributes['storage'] <> 'warm';
```

### Actual Result

```
ColumnUnknownException[Column attributes['storage'] unknown]
```

### Expected Result

A non-empty result, including a single metadata record.

**Comment by amotl** [comment]
When using `SET error_on_unknown_object_key = 'false';` before running the metadata query, it works.

**Comment by jeeminso** [comment]
Thanks for reporting @amotl .

Since `attribute` object from `sys.nodes` is `dynamic`, this is expected behaviour and need to set `error_on_unknown_object_key = 'false'`. Just wondering if it makes sense that system tables' objects' column policies are `dynamic`.

**Comment by jeeminso** [comment]
Actually, I was wrong about setting `error_on_unknown_object_key = 'false'`. `attributes['storage']` exists and we cannot resolve the sub-column:

```
cr> create table t01 (id INT PRIMARY KEY ) partitioned by ("id") WITH ("routing.allocation.require.storage" = 'hot');
 
CREATE OK, 1 row affected (0.127 sec)
cr> insert into t01 (id) values (42);
 
INSERT OK, 1 row affected (1.802 sec)
cr> select attributes from sys.nodes;
+--------------------+
| attributes         |
+--------------------+
| {"storage": "hot"} |
+--------------------+
SELECT 1 row in set (0.091 sec)
cr> select attributes['storage'] from sys.nodes;
ColumnUnknownException[Column attributes['storage'] unknown]
```
=> systemTable.getReference() cannot find `attributes['storage']`.

**Comment by amotl** [comment]
Is it related to this change, merged yesterday?

- https://github.com/crate/crate/pull/17052

**Comment by amotl** [comment]
Thank you for coming up with a fix so quickly.

**Comment by jeeminso** [comment]
Thank you @amotl , this regression is fixed by #17103.
